src/effects/DynamicRangeProcessorEditor.cpp

```diff
@@ -0,0 +1,212 @@
+#include "DynamicRangeProcessorEditor.h"
+#include "AllThemeResources.h"
+#include "CompressorProcessor.h"
+#include "DynamicRangeProcessorHistoryPanel.h"
+#include "DynamicRangeProcessorOutputs.h"
+#include "EffectInterface.h"
+#include "ShuttleGui.h"
+#include "Theme.h"
+#include "widgets/LinearDBFormat.h"
+#include "widgets/RulerPanel.h"
+#include "widgets/valnum.h"
+#include <wx/slider.h>
+#include <wx/textctrl.h>
+
+namespace
+{
+constexpr auto historyPanelId = wxID_HIGHEST + 1;
+constexpr auto rulerPanelId = wxID_HIGHEST + 2;
+
+float GetDbRange(float maxCompressionDb)
+{
+   constexpr auto minRange = 3.f;
+   return std::max(minRange, maxCompressionDb / 2);
+}
+
+void AddHistoryPanel(
+   wxWindow* parent, const DynamicRangeProcessorHistoryUPtr& history,
+   DynamicRangeProcessorOutputs& outputs, EffectSettingsAccess& access)
+{
+   const CompressorSettings* settings = access.Get().cast<CompressorSettings>();
```

Comment:
I'm not sure that this would work - not that I wouldn't like it to!
I think it's because of this guy: 
![image](https://github.com/audacity/audacity/assets/22740106/6a686b28-d2a8-403b-b79d-950a8b6ac487)
The `EffectSettingsAccess` class initially creates a default-constructed struct. Creating a new type is the only way I found of circumventing that.

Reply:
Hm, didn't think of it. `EffectWithSettings<>::MakeSettings` is virtual and can be overriden, will that help?